src/amo/components/PermissionsCard/permissions.js

```diff
@@ -0,0 +1,237 @@
+    }
+    return result;
+  }
+
+  // Format and sequence all the Permission components.
+  formatPermissions(addonPermissions: Array<string>): Array<Object> {
+    const permissionsToDisplay = [];
+    const permissions = { hosts: [], permissions: [] };
+
+    // First, categorize them into host permissions and regular permissions.
+    for (const permission of addonPermissions) {
+      const { type, value } = this.classifyPermission(permission);
+      permissions[type].push(value);
+    }
+
+    // Classify the host permissions.
+    let allUrls = false;
+    const wildcards = [];
+    const sites = [];
+    for (const permission of permissions.hosts) {
+      if (permission === '<all_urls>') {
+        allUrls = true;
+        break;
+      }
+      if (permission.startsWith('moz-extension:')) {
+        // eslint-disable-next-line no-continue
+        continue;
+      }
+      const match = /^[a-z*]+:\/\/([^/]+)\//.exec(permission);
+      if (!match) {
+        log.debug(`Host permission string "${permission}" appears to be invalid.`);
+        // eslint-disable-next-line no-continue
+        continue;
+      }
+      if (match[1] === '*') {
+        allUrls = true;
+      } else if (match[1].startsWith('*.')) {
+        wildcards.push(match[1].slice(2));
+      } else {
+        sites.push(match[1]);
+      }
+    }
+
+    // Format the host permissions.  If we have a wildcard for all urls,
+    // a single string will suffice.  Otherwise, show domain wildcards
+    // first, then individual host permissions.
+
+    if (allUrls) {
+      permissionsToDisplay.push(
+        <Permission
+          className="hostPermission"
+          description={this.permissionStrings.allUrls}
+          key="allUrls"
+        />
+      );
+    } else {
+      // Formats a list of host permissions.  If we have 4 or fewer, display
+      // them all, otherwise display the first 3 followed by an item that
+      // says "...plus N others".
+      const formatHostPermissions = (list, itemKey, moreKey) => {
+        const formatItems = (items) => {
+          permissionsToDisplay.push(...items.map((item) => {
+            return (
+              <Permission
+                className="hostPermission"
+                description={this.permissionStrings[itemKey](item)}
+                key={item}
+              />
+            );
+          }));
+        };
+        if (list.length < 5) {
+          formatItems(list);
+        } else {
+          formatItems(list.slice(0, 3));
+
+          const remaining = list.length - 3;
+          permissionsToDisplay.push(
+            <Permission
+              className="hostPermission"
+              description={this.permissionStrings[moreKey](remaining)}
+              key={moreKey}
+            />
+          );
+        }
+      };
+
+      formatHostPermissions(wildcards, 'wildcard', 'tooManyWildcards');
+      formatHostPermissions(sites, 'oneSite', 'tooManySites');
```

Comment:
I see "Access your data in 3 other domains" and "Access your data on 2 other sites" for the [same extension](http://localhost:3000/en-US/firefox/addon/all-of-the-permissions/). This seems redundant. What about combining this message into one? It could be filed a follow-up issue if it's too much work.


<img width="301" alt="screenshot 2018-03-08 12 15 26" src="https://user-images.githubusercontent.com/55398/37168883-ddeab6d0-22cb-11e8-82fa-0b92a0436120.png">

Reply:
This is what Firefox does (I'm pretty sure) and I tried to match what Firefox does as much as possible. It does make sense that we don't really need both of those messages, although there is a difference between `sites` and `domains`. It's not possible for the user to differentiate when they both just say `on x other ...` though, granted.

@devaneymoz Do you have an opinion on this?

Reply:
FTR, I confirmed that the above is what Firefox shows. It does not combine "3 other domains" and "2 other sites" into "5 other x".

Reply:
ok, I see the same thing in Firefox. I don't really see how it's helpful but I suppose it's fine to match the Firefox UI, at least for a first pass.

<img width="379" alt="screenshot 2018-03-08 14 09 18" src="https://user-images.githubusercontent.com/55398/37174005-e78b286e-22da-11e8-9ad8-f5c12e84efb7.png">